IPython/core/ultratb.py

```diff
@@ -1094,12 +1094,14 @@ def structured_traceback(self, etype, evalue, etb, tb_offset=None,
                 etype, evalue, etb = exception
             else:
                 evalue = None
+            chained_exc_ids = set()
```

Comment:
Ah, I see what you mean. I think it still won't show you the cycle if it looks like `A-B-C-D-B-...`, but I don't think that's especially important. The main thing is to make sure it doesn't go into an infinite loop. Thanks!